app/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/PollingNewMessagesWithWorkItemJob.scala

```diff
@@ -104,22 +102,27 @@ class PollingNewMessageWithWorkItemJob @Inject()
       }
   }
 
-
-  private def getNewMessages(exciseNumber: String): Future[(Boolean, NewMessageResult)] = {
+  private def getNewMessages(exciseNumber: String): Future[NewMessageResult] = {
     newMessageService.getNewMessagesAndAcknowledge(exciseNumber)
       .flatMap {
         case Some((response, messageCount)) if messageCount > 10 && response.message.nonEmpty =>
-          saveToDB(exciseNumber, response).map(result => (result, MoreMessagesToGet))
+          saveToDB(exciseNumber, response).map(_ => MoreMessagesToGet)
 
         case Some((response, _)) if response.message.nonEmpty =>
-          saveToDB(exciseNumber, response).map(result => (result, MessageReceived))
+          saveToDB(exciseNumber, response).map(_ => Processed)
+
+        case Some((_, _)) =>
+          // No messages returned
+          Future.successful(Processed)
 
-        case _ => Future.successful((true, NoMessageFound))
+        case _ =>
```

Comment:
Tweaked the logic but the response from the EIS is non-empty even when they are no messages available so it ends up in the case of Some(response, _)

[This can get confusing because we have Message meaning multiple things here. Will try and clear that up]